### PR TITLE
Update 1.5-disks-setup to be more general purpose.

### PR DESCRIPTION
**Pull request by rtamalin**
templates/volume_groups.j2
==========================
Simplify the template used to volume_groups variable and eliminate the
unnecessary diskcount handling.

vars/disks_config.yml
=====================
Move the logical volumes configuration specification to a top-level
vars file that also includes playbook tier and node tier association,
and an optional fstype specification. Dynamically determine options
string for the the LV creation based on optional stripesize setting.
Support optional fsopts setting to be used when formatting file
systems.

Misc Changes
============
Implement equivalent changes to #1166 so that VGs, LVs and associated
file systens are only created when playbook tier is sapos.

Ensure the that tier set_fact action is always run, by tagging it with
always in the relevant playbooks.

TODO
====
  * Potentially add file system mount point, and possibly nfs mount
    point specification to the disks_config.yml
  * Add support for specifying the tier in which the LVs will be
    created, and the tier in which the LVs will be mounted, to the
    the LV entries in disks_config.yml
  * Support specification of LVs associated with VG consuming extra
    space on the OS disk to be created/formated/mounted during the
    os tier playbook run.

### CODE REVIEW

**Comment by rtamalin** [comment]
Note that this is a draft PR that depends upon the unmerged changes in PRs #1160 and #1163 so once those PRs have reviewed/updated and merged/closed I can rebase this PR, applying any relevant changes.

**Review by KimForss — COMMENTED** [review]
Can you rebase as I approved and merged the tags

**Comment by rtamalin** [comment]
Rebased onto latest alfa. Was having issues getting a system deployed that would allow me to test it.

File: deploy/ansible/playbook_05_02_sap_pas_install.yaml [hunk 1]
```diff
@@ -25,7 +25,9 @@
 #
 # -------------------------------------+---------------------------------------8
   - set_fact:
-      tier:     pas                                                          # SAP Install phase: PAS - Primary Application Server
+      tier:     pas                                                          # SAP Install phase: DBLOAD
```
Comment (KimForss) [inline deploy/ansible/playbook_05_02_sap_pas_install.yaml:28]: Keep the old comment: # SAP Install phase: PAS - Primary Application Server

Comment (rtamalin) [reply deploy/ansible/playbook_05_02_sap_pas_install.yaml:28]: That is a weird one as I don't think I touched that line in my original editing... Possibly a git merge/rebase glitch at some point where I kept an older version of the line.k

Comment (rtamalin) [reply deploy/ansible/playbook_05_02_sap_pas_install.yaml]: Fixed in latest version

**Comment by rtamalin** [comment]
@KimForss I have fixed the issue you spotted and also updated the `disk_type_to_name_map` entries as we discussed yesterday.

**Review by KimForss — APPROVED** [review]
looks good to me

**Pull request merged**